app/controllers/application_controller.rb

```diff
@@ -25,8 +25,8 @@ def append_info_to_payload(payload)
     payload[:ip] = request.remote_ip
   end
 
-  def analytics
-    @analytics ||= Analytics.new(current_user, request_attributes, ahoy)
+  def analytics(service = Analytics.new(current_user, request))
```

Comment:
On second thought, no. The memoization is needed for the tests. I can't think of any scenario where we would want to call `analytics` with different parameters in the same request.